code/game/objects/machinery/miner.dm

```diff
@@ -0,0 +1,167 @@
+		playsound(loc, 'sound/items/wirecutter.ogg', 25, 1)
+		user.visible_message("<span class='notice'>[user] starts securing [src]'s wiring.</span>",
+		"<span class='notice'>You start securing [src]'s wiring.</span>")
+		if(!do_after(user, 120, TRUE, src, BUSY_ICON_BUILD) || miner_status != MINER_MEDIUM_DAMAGE)
+			return FALSE
+		playsound(loc, 'sound/items/wirecutter.ogg', 25, 1)
+		miner_status = MINER_SMALL_DAMAGE
+		user.visible_message("<span class='notice'>[user] secures [src]'s wiring.</span>",
+		"<span class='notice'>You secure [src]'s wiring.</span>")
+		update_icon()
+		return TRUE
+
+/obj/machinery/miner/wrench_act(mob/living/user, obj/item/I)
+	if(miner_status == MINER_SMALL_DAMAGE)
+		if(user.skills.getRating("engineer") < SKILL_ENGINEER_ENGI)
+			user.visible_message("<span class='notice'>[user] fumbles around figuring out [src]'s tubing and plating.</span>",
+			"<span class='notice'>You fumble around figuring out [src]'s tubing and plating.</span>")
+			var/fumbling_time = 10 SECONDS - 2 SECONDS * user.skills.getRating("engineer")
+			if(!do_after(user, fumbling_time, TRUE, src, BUSY_ICON_UNSKILLED))
+				return FALSE
+		playsound(loc, 'sound/items/ratchet.ogg', 25, 1)
+		user.visible_message("<span class='notice'>[user] starts repairing [src]'s tubing and plating.</span>",
+		"<span class='notice'>You start repairing [src]'s tubing and plating.</span>")
+		if(do_after(user, 150, TRUE, src, BUSY_ICON_BUILD) && miner_status == MINER_SMALL_DAMAGE)
+			playsound(loc, 'sound/items/ratchet.ogg', 25, 1)
+			miner_status = MINER_RUNNING
+			user.visible_message("<span class='notice'>[user] repairs [src]'s tubing and plating.</span>",
+			"<span class='notice'>You repair [src]'s tubing and plating.</span>")
+			update_icon()
+			return TRUE
+
+/obj/machinery/miner/examine(mob/user)
+	..()
+	if(ishuman(user))
+		switch(miner_status)
+			if(MINER_DESTROYED)
+				to_chat(user, "<span class='info'>It's heavily damaged, and you can see internal workings.</span>")
+				to_chat(user, "<span class='info'>Use a blowtorch, then wirecutters, then wrench to repair it.</span>")
+			if(MINER_MEDIUM_DAMAGE)
+				to_chat(user, "<span class='info'>It's damaged, and there are broken wires hanging out.</span>")
+				to_chat(user, "<span class='info'>Use a wirecutters, then wrench to repair it.</span>")
+			if(MINER_SMALL_DAMAGE)
+				to_chat(user, "<span class='info'>It's lightly damaged, and you can see some dents and loose piping.</span>")
+				to_chat(user, "<span class='info'>Use a wrench to repair it.</span>")
+			if(MINER_RUNNING)
+				to_chat(user, "<span class='info'>[src]'s internal storage currently has [stored_mineral] sheets stored.</span>")
+
+/obj/machinery/miner/attack_hand(mob/living/user)
+	var/mob/living/L = user
+	if(miner_status != MINER_RUNNING)
+		to_chat(L, "<span class='warning'>[src] is damaged!</span>")
+		return
+	if(stored_mineral != 0)
+		if(stored_mineral >= 50)	//We shouldnt get more, but just in case
+			stop_processing()

```

Comment:
wouldn't this already be too late?

Reply:
it's supposed to be start_processing, woops